Dockerfile

```diff
@@ -0,0 +1,55 @@
+FROM ubuntu:18.04
+
+LABEL maintainer="Dan Levin <dan@badpacket.in>, Martin Häcker <spamfaenger@gmx.de>"
+LABEL description="This image is intended as a developent environment for task-tracker."
+
+ENV LC_ALL=C.UTF-8 LANG=C.UTF-8  
+
+# How does docker figure out when to rerun these lines?
+# FIXME according to https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache
+# The cache for this RUN action is only invalidated if the line changes. So how do I ensure to either get updates regularly
+# or force the build process to respect some minimum or pinned versions of packages?
+RUN apt-get update \
+    && apt-get -y install \
+        python3.7 \
+        python3-pip \
+        python3-venv \
+        curl \
+        git \
+        gnupg2
+
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+    && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
+    && apt-get update \
+    && apt-get -y install yarn
+
+WORKDIR /task-tracker
+
+# JS dependencies
+COPY package.json yarn.lock /tmp/
+RUN cd /tmp && yarn install
+# TODO consider how much of this could be run on a different / dedicated container
```

Comment:
you might benefit from using a multi-stage build, but it's not clear to me at the moment. multi-stage will not speed up your build, but rather reduce the final image size by allowing you to pick specific artifacts from a throw-away image for inclusion into the final image.